akka-persistence/src/main/scala/akka/persistence/PersistentView.scala

```diff
@@ -361,6 +361,10 @@ trait PersistentView extends Actor with Snapshotter with Stash with StashFactory
     override def recoveryRunning: Boolean = false
 
     override def stateReceive(receive: Receive, message: Any): Unit = message match {
+      case ReplayedMessage(p) ⇒
+        // we can get ReplayedMessage here if it was stashed by user during replay
+        // unwrap the payload
+        PersistentView.super.aroundReceive(receive, p.payload)
```

Comment:
>hmm, is there a valid use case for stashing replayed messages?

In our use case we are using a PersistentView to create a (denormalised) read model. Most of your implementations require no stashing - just mapping logic. However, sometimes we join replayed events with some data before writing to the read model. That data might not be available during event consumption (due to eventual consistency, load or a network failure). So we stash all incoming events for a few seconds, then unstash and check for availability of that data again. We repeat that until the data is available.

So basically we can delay event consumption with stash/unstash in a non-blocking fashion.

I am aware this could lead to OOMs but given the right circumstances (low event publishing frequency) it works very well.